### PR TITLE
sys: simplify shell

### PR DESCRIPTION
**Pull request by kaspar030**
seeing multiple different types of getchar/putchar given to shell, with no actual effect as all output goes to the default stdio anyways, I've decided to get rid of the parameters.

If we one day want shells with IO other than stdio, we should implement it right and redirect stdio.

~~This is breaks all platforms not offering getchar/putchar. (WIP because of this)~~

### CODE REVIEW

File: examples/rpl_udp/main.c [hunk 1]
```diff
@@ -43,14 +41,12 @@ int main(void)
...
-    /* start shell */
-    posix_open(uart0_handler_pid, 0);
     net_if_set_src_address_mode(0, NET_IF_TRANS_ADDR_M_SHORT);
     id = net_if_get_hardware_address(0);
 
-    shell_t shell;
-    shell_init(&shell, shell_commands, UART0_BUFSIZE, uart0_readc, uart0_putc);
+    /* start shell */
+    char line_buf[SHELL_DEFAULT_BUFSIZE];
+    shell_run(shell_commands, line_buf, SHELL_DEFAULT_BUFSIZE);
```
Comment (haukepetersen) [inline examples/rpl_udp/main.c]: where is `line_buf` defined?

**Comment by haukepetersen** [comment]
very nice, +1!

File: examples/rpl_udp/main.c (hunk 1 above)
Comment (kaspar030) [inline examples/rpl_udp/main.c]: uhm, one line above? :)

**Comment by kaspar030** [comment]
(I mark the PR "ready for CI" to see build errors)

**Comment by haukepetersen** [comment]
only some :-)

**Comment by kaspar030** [comment]
- rebased
- fixed chronos fix

**Comment by haukepetersen** [comment]
could you pls squash?

**Comment by kaspar030** [comment]
- rebased
- squashed

File: sys/include/shell.h [hunk 2]
```diff
@@ -29,6 +29,11 @@ extern "C" {
 #endif
 
 /**
+ * @brief Default shell buffer size (maximum line length shell can handle)
+ */
+#define SHELL_DEFAULT_BUFSIZE   (128)
```
Comment (OlegHahm) [inline sys/include/shell.h]: Shouldn't be 64 bytes enough?

**Comment by OlegHahm** [comment]
Shell doesn't work on MSB-A2.

File: sys/include/shell.h (hunk 2 above)
Comment (haukepetersen) [inline sys/include/shell.h]: I would go with 128, its the safer bet...

Comment (OlegHahm) [inline sys/include/shell.h]: Ok, if I think of commands like `fibroute` you're probably right.

**Comment by haukepetersen** [comment]
#3530 should fix everything for the msba2

**Comment by OlegHahm** [comment]
Ok, works on Z1 and pba-d-01-kw2x. Let's wait for #3530 then.

**Comment by haukepetersen** [comment]
reviewing and testing it as we speak

**Comment by kaspar030** [comment]
breaks msb430.

**Comment by haukepetersen** [comment]
Will look into the msb430 anyway, so I try to deal with it.

**Comment by kaspar030** [comment]
- rebased

**Comment by haukepetersen** [comment]
MSP430 periph driver are under way, hope we can merge this this week...

**Comment by cgundogan** [comment]
needs a rebase, because of the `ng_`=> `gnrc_` renaming

**Comment by kaspar030** [comment]
- rebased

**Comment by kaspar030** [comment]
-rebased

**Comment by kaspar030** [comment]
all dependencies are in!

- rebased

**Comment by haukepetersen** [comment]
code looks good, works on every board I tested (i.e `msb-430h`, `nucleo-l1`, `arduino-due`, `avsextrem`, `msba2`, `yunjia-nrf`) -> ACK

**Comment by haukepetersen** [comment]
just for fun, also successfully tested for the `frdm-k64f` and the `pba-d-01-kw2x`

**Comment by OlegHahm** [comment]
ACK

**Comment by haukepetersen** [comment]
and go!

**Comment by OlegHahm** [comment]
Nice work, @kaspar030!

**Comment by kaspar030** [comment]
Thanks, and thanks for reviewing!

**Pull request merged**